docs/dev.rst

```diff
@@ -1,20 +1,522 @@
-Develop a project
+Develop a Project
 =================
 
-Set up a dev org
+A general overview on how to develop a Salesforce project with CumulusCI.
+
+
+
+Set Up a Dev Org
 ----------------
 
-Track and retrieve changes
---------------------------
+The ``dev_org`` flow creates an org to develop on by moving all metadata (managed and unmanaged) into the org, and configuring it to be ready for development.
+
+.. note:: Run ``cci flow info dev_org`` for a full list of the ``dev_org`` flow steps.
+
+To run the ``dev_org`` flow against the project's `default org<TODO>`_:
+
+.. code-block:: console
+
+    $ cci flow run dev_org
+
+You can also use the ``--org`` option to explicitly list the org.
+
+    Example: Run the ``dev_org`` flow against the org currently defined as ``dev`` in CumulusCI.
+
+.. code-block:: console
+
+    $ cci flow run dev_org --org dev
+
+..
+
+    Open the new ``dev`` org to begin development.
+
+.. code-block:: console
+
+    $ cci org browser dev
+
+
+Set the Capture State
+^^^^^^^^^^^^^^^^^^^^^
+
+To make changes to capture in an org, use the ``snapshot_changes`` task to create a snapshot. A snapshot tells the Salesforce CLI source tracking to set the org's current state as a baseline for changes to be made against.
```

Comment:
I don’t fully understand what a snapshot is/does. What does a snapshot look like exactly? Should we show an example here?

Reply:
A snapshot is a "point-in-time" capturing of an org's current state. It is essentially telling the Salesforce CLI, "Hey there, I've gone and mucked with some stuff in this org, but the next time I ask you for all of the changes I've made in this org, please ignore everything before this point in time."

This why most org flows end with the `snapshot_changes` task; so that when a person wants to retrieve the work they've done in this org back into their local github repo, they don't get all of the changes that were made when setting up the org.

Reply:
@Br4nd0R Interesting. And where does one examine a snapshot? CLI? GitHub?

Reply:
@justinvholt - I'm not positive how the actual snapshot is stored as the snapshots are handled by the Salesforce CLI. (There are commands for listing (sfdx force:org:snapshot:list) snapshots and getting information on (sfdx force:org:snapshot:get) a single snapshot.

There shouldn't be a reason for users to need to "view" a snapshot. This is just a concept that is useful to solidify for developing on salesforce scratch org.

Reply:
@Br4nd0R The snapshot being referred to in the context of the list_changes/retrieve_changes/snapshot_changes tasks is a completely different thing than the snapshot in the context of the sfdx force:org:snapshot commands. cci's snapshot is just a file storing a list of what the most recent known revision of each component is. sfdx's snapshot is a Pilot feature for creating a scratch org by copying another org.